File: solutions/cpp/second_largest.cpp
#include <iostream>
#include <set>
#include <vector>
using namespace std;

int secondLargest(vector<int>& arr) {
    set<int> s(arr.begin(), arr.end());
    auto it = s.end(); 
    it--; 
    it--; 
    return *it;
}

int main() {
    vector<int> v = {10,20,4,45,99};
    cout << secondLargest(v);
}

File: solutions/java/SecondLargest.java
import java.util.*;

public class SecondLargest {
    public static int secondLargest(int[] arr) {
        Set<Integer> set = new HashSet<>();
        for (int n : arr) set.add(n);
        java.util.List<Integer> list = new java.util.ArrayList<>(set);
        java.util.Collections.sort(list);
        return list.get(list.size()-2);
    }
    public static void main(String[] args) {
        System.out.println(secondLargest(new int[]{10,20,4,45,99}));
    }
}